metadata-ingestion/src/datahub/ingestion/source/tableau/tableau.py

```diff
@@ -255,6 +255,46 @@ def get_tableau_auth(
             )
         return authentication
 
+    def check_user_role(
+        self, server: Server
+    ) -> Dict[Union[SourceCapability, str], CapabilityReport]:
+        assert server.user_id, "make the connection with tableau"
+
+        capability_dict: Dict[Union[SourceCapability, str], CapabilityReport] = {
+            "metadataRole": CapabilityReport(
+                capable=True,
+            )
+        }
+
+        failure_reason: str = (
+            "The user does not possess the `Site Administrator Explorer` role."
+        )
+
+        mitigation_message: str = "Assign `Site Administrator Explorer` role to the user {}. Refer setup guide: https://datahubproject.io/docs/quick-ingestion-guides/tableau/setup"
+
+        try:
+            user = server.users.get_by_id(server.user_id)
```

Comment:
let's add a helper that gets the site role from the user first

I'd like to add that to the ingestion report, and ideally also show a warning during ingestion if we don't have the expected permissions - that way this will be clear even if they don't click "test connection"

Reply:
done